pkg/cmd/roachtest/tests/backup_fixtures.go

```diff
@@ -360,6 +363,92 @@ func (bd *backupDriver) compactionJobStates(
 	return successes, running, failures, nil
 }
 
+// fingerprintFixture computes a fingerprint of the fixture. It first takes an
+// incremental backup to ensure that the database state matches the fixture.
+// Note: This assumes that the workload has been stopped so that the state of
+// the database matches the fixture contents.
+func (bd *backupDriver) fingerprintFixture(ctx context.Context) string {
+	conn := bd.c.Conn(ctx, bd.t.L(), 1)
+	defer conn.Close()
+	sql := sqlutils.MakeSQLRunner(conn)
+
+	// Take one more backup to ensure database state matches the fixture for
+	// fingerprinting. Again, this assumes the workload has been stopped.
+	bd.t.L().Printf("taking incremental backup before fingerprinting")
+	bd.takeIncrementalBackup(ctx, sql)
+
+	return fingerprintDatabase(ctx, bd.t, sql, bd.sp.fixture.DatabaseName())
+}
+
+// takeIncrementalBackup synchronously takes an incremental backup.
+func (bd *backupDriver) takeIncrementalBackup(ctx context.Context, sql *sqlutils.SQLRunner) {
+	collectionURI := bd.registry.URI(bd.fixture.DataPath)
+	backupStmt := fmt.Sprintf(
+		`BACKUP DATABASE %s INTO LATEST IN '%s'`, bd.sp.fixture.DatabaseName(), collectionURI.String(),
+	)
+	sql.Exec(bd.t, backupStmt)
+}
+
+// fingerprintDatabase computes a fingerprint of the database by fingerprinting
+// each table in the database and combining the fingerprints with an XOR.
+func fingerprintDatabase(
+	ctx context.Context, t test.Test, sql *sqlutils.SQLRunner, db string,
+) string {
+	tables := getDatabaseTables(ctx, t, sql, db)
+	var wg sync.WaitGroup
+	wg.Add(len(tables))
+
+	var fingerprint int64
+	var mu syncutil.Mutex
+	for _, table := range tables {
+		table := table
+		go func() {
+			defer wg.Done()
+
+			rows := sql.Query(t, fmt.Sprintf(
+				"SELECT fingerprint FROM [SHOW EXPERIMENTAL_FINGERPRINTS FROM TABLE %s]", table,
+			))
+			defer rows.Close()
+
+			var tableFingerprint int64
+			for rows.Next() {
+				var indexFingerprint int64
+				err := rows.Scan(&indexFingerprint)
+				require.NoError(t, err, "error scanning fingerprint for table %s", table)
+				tableFingerprint ^= indexFingerprint
+			}
+
+			mu.Lock()
+			defer mu.Unlock()
+			fingerprint ^= tableFingerprint
```

Comment:
might be nice to store map of table name to fingerprint instead of the aggregated fingerprint.

Reply:
Good point, no need to reduce the granularity here. If we're taking that step, store a map from table name to another map mapping index name to fingerprint?